modules/http/http.c

```diff
@@ -202,20 +202,40 @@ _add_message_to_batch(HTTPDestinationWorker *self, LogMessage *msg)
 }
 
 static worker_insert_result_t
-_map_http_status_to_worker_status(glong http_code)
+_map_http_status_to_worker_status(HTTPDestinationWorker *self, glong http_code)
 {
+  HTTPDestinationDriver *owner = (HTTPDestinationDriver *) self->super.owner;
   worker_insert_result_t retval;
 
   switch (http_code/100)
     {
-    case 4:
-      retval = WORKER_INSERT_RESULT_DROP;
+    case 1:
+    case 2:
+      retval = WORKER_INSERT_RESULT_SUCCESS;
       break;
-    case 5:
+    case 3:
+      msg_notice("Server returned with a 3XX (redirect) status code, which was not handled by curl. "
+                 "Either accept-redirect() is set to no, or this status code is unknown. Trying again",
+                 evt_tag_str("url", owner->url),
+                 evt_tag_int("status_code", http_code),
+                 log_pipe_location_tag(&owner->super.super.super.super));
+      retval = WORKER_INSERT_RESULT_ERROR;
+      break;
+    case 4:
+      msg_notice("Server returned with a 4XX (client errors) status code, which means we are not "
+                 "authorized or the URL is not found. Trying again",
+                 evt_tag_str("url", owner->url),
+                 evt_tag_int("status_code", http_code),
+                 log_pipe_location_tag(&owner->super.super.super.super));
       retval = WORKER_INSERT_RESULT_ERROR;
       break;
     default:
```

Comment:
Cosmetic, but I would prefer default statement to be the last case.

Reply:
the default falls through to the 5XX case that's why it was in that order. But I've added a followup patch, that among other things makes "default" the last one.